src/components/selectionBoxLayer.ts

```diff
@@ -183,9 +210,33 @@ export module Components {
      *
      * Returns an undefined array if the edges are not backed by a scale.
      */
-    public xExtent(): (number | { valueOf(): number })[] {
+    public xExtent(): (number | { valueOf(): number })[];
+    /**
+     * Sets the data values backing the left and right edges of the box.
+     */
+    public xExtent(xExtent: (number | { valueOf(): number })[]): SelectionBoxLayer;
+    public xExtent(xExtent?: (number | { valueOf(): number })[]): any {
       // Explicit typing for Typescript 1.4
-      return [this._boxLeftDataValue, this._boxRightDataValue];
+      if (xExtent == null) {
+        return this._getXExtent();
+      }
+      this._setXExtent(xExtent);
+      this._xBoundsMode = PropertyMode.VALUE;
+      this.render();
+      return this;
+    }
+
+    private _getXExtent(): (number | { valueOf(): number })[] {
+      return this._xBoundsMode === PropertyMode.VALUE ?
+               this._xExtent :
```

Comment:
I don't really like this sort of indentation because it can easily get off sync when global search replacing for var renames